src/Datasource/Exception/PageOutOfBoundsException.php

```diff
@@ -1,28 +1,11 @@
 <?php
 declare(strict_types=1);
 
-/**
- * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
- * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
- *
- * Licensed under The MIT License
- * Redistributions of files must retain the above copyright notice.
- *
- * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
- * @since         3.5.0
- * @license       http://www.opensource.org/licenses/mit-license.php MIT License
- */
-namespace Cake\Datasource\Exception;
-
-use Cake\Core\Exception\CakeException;
-
-/**
- * Exception raised when requested page number does not exist.
- */
-class PageOutOfBoundsException extends CakeException
-{
-    /**
-     * @inheritDoc
-     */
-    protected $_messageTemplate = 'Page number %s could not be found.';
-}
+class_alias(
```

Comment:
If I remember correctly, this needs to go into the new file and class_exists() should go here instead to force loading.

If we don't do that, then the alias won't exist when we `new PageOutOfBoundsException()` and user code won't catch it since they're using the alias.

Reply:
For other aliasing we've put the `class_alias()` in the *old*  filename and alias to the new name. At the end of the old file composer will load the new name.

https://github.com/cakephp/cakephp/blob/806a9a5016a3abb8ccbc9a2d9c77bc721705e947/src/Database/Type.php

is an aliasing we currently have in 4.x

Reply:
This is a subtle issue where user code is using a class name in a catch() but not instantiating the object. Without php loading the class, the alias won't exist.

Reply:
@othercorey I have now done what you did for `Core\Exception` => `Core\CakeException` rename. Please check.

Unfortunate thing about that it we can't show a deprecation warning for the old name, since it's always loaded when loading the new class.

Reply:
Yea, too bad we can't deprecate a class name. Looks ok to me.